fix: fix mismatch of to Enum function name

dorotaphanSiili opened the pull request:
<!-- 
Thanks for creating this pull request 🤗

Please make sure that the pull request is limited to one type (docs, feature, etc.) and keep it as small as possible. You can open multiple prs instead of opening a huge one.
-->

<!-- If this pull request closes an issue, please mention the issue number below -->
Closes # <!-- Issue # here -->

## 📑 Description
<!-- Add a brief description of the pr -->

<!-- You can also choose to add a list of changes and if they have been completed or not by using the markdown to-do list syntax
- [ ] Not Completed
- [x] Completed
-->

## ✅ Checks
<!-- Make sure your pr passes the CI checks and do check the following fields as needed - -->
- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

## ℹ Additional Information
<!-- Any additional information like breaking changes, dependencies added, screenshots, comparisons between new and old behavior, etc. -->

dorotaphanSiili committed:
ci: remove installing clang

dorotaphanSiili committed:
fix: fix mismatch of to Enum function name

w4bremer commented on templates/module/generated/api/datastructs.api.cpp.tpl:
```diff
@@ -12,7 +12,7 @@ namespace {{ Camel .Module.Name }} {
 // Enumeration {{$class}}
 // ********************************************************************
-{{$class}}Enum to{{Camel $class}}Enum(std::uint8_t v, bool *ok)
+{{$class}}Enum to{{upper1 $class}}Enum(std::uint8_t v, bool *ok)
```
Please add a test bed api test case for this.

dorotaphanSiili committed:
update to new test-api

dorotaphanSiili committed:
fix(olink): empty interface fix

For interfaces wihtout the properties and signals
one of members for service adapter is never used.
The in-parameter stayed although it is not necessary, for consisntency
and compatibility.
The not necessary member was removed.
